### PR TITLE
small tweaks

### PR DESCRIPTION
**Pull request by bryzettler**

### CODE REVIEW

File: utils/uiTypes/VotePlugin.ts [hunk 1]
```diff
@@ -199,6 +198,7 @@ export class VotingClient {
     }
 
     if (this.client instanceof HeliumVsrClient) {
+      console.log('WTF WHEN')
```
Comment (bryzettler) [inline utils/uiTypes/VotePlugin.ts:201]:
```suggestion
```

File: components/VotePanel/CastVoteButtons.tsx [hunk 2]
```diff
@@ -70,6 +94,56 @@ export const CastVoteButtons = () => {
...
+        )
+      } catch (ex) {
+        if (isNftPlugin) {
+          closeNftVotingCountingModal(
+            (client.client as unknown) as NftVoterClient,
+            proposal!,
+            wallet!.publicKey!
+          )
+        }
+        //TODO: How do we present transaction errors to users? Just the notification?
```
Comment (asktree) [inline components/VotePanel/CastVoteButtons.tsx:139]: sendTransactionsV3 does it

File: components/VotePanel/CastVoteButtons.tsx [hunk 3]
```diff
@@ -70,6 +94,56 @@ export const CastVoteButtons = () => {
...
   const isInCoolOffTime = useIsInCoolOffTime()
 
+  const handleClick = async (vote: 'yes' | 'no') => {
+    setVote(vote)
+    setSubmitting(true)
+
+    if (allowDiscussion) {
+      setShowVoteModal(true)
+    } else {
+      const confirmationCallback = async () => {
```
Comment (asktree) [inline components/VotePanel/CastVoteButtons.tsx:104]: so is this just duplicated code from the vote modal ?

Comment (asktree) [reply components/VotePanel/CastVoteButtons.tsx:104]: if so lets move it to a shared hook

File: components/chat/DiscussionPanel.tsx [hunk 4]
```diff
@@ -1,12 +1,22 @@
+import React, { useMemo } from 'react'
 import DiscussionForm from './DiscussionForm'
 import Comment from './Comment'
 import useWalletStore from '../../stores/useWalletStore'
 
-const DiscussionPanel = () => {
+const DiscussionPanel: React.FC<any> = () => {
```
Comment (asktree) [inline components/chat/DiscussionPanel.tsx:6]: ?

File: components/chat/DiscussionPanel.tsx [hunk 5]
```diff
@@ -1,12 +1,22 @@
...
   const { chatMessages, voteRecordsByVoter, proposalMint } = useWalletStore(
     (s) => s.selectedProposal
   )
 
+  const sortedMessages = useMemo(
```
Comment (asktree) [inline components/chat/DiscussionPanel.tsx:11]: wise

File: hooks/useRealm.tsx [hunk 6]
```diff
@@ -244,6 +244,10 @@ export default function useRealm() {
     realmInfo,
     symbol,
     voteSymbol: realmInfo?.voteSymbol,
+    allowDiscussion:
```
Comment (asktree) [inline hooks/useRealm.tsx:247]: `useRealm` already surfaces `realmInfo`, there is no need to create a new field. I don't like the existing way that fields are constantly re-exported like this and I'm in the process of killing it.

**Review by asktree — CHANGES_REQUESTED** [review]
left some comments.

**Review by asktree — CHANGES_REQUESTED** [review]
left some comments.

File: components/VotePanel/CastVoteButtons.tsx [hunk 7]
```diff
@@ -58,6 +67,21 @@ const useCanVote = () => {
...
+  const wallet = useWalletOnePointOh()
+  const connection = useWalletStore((s) => s.connection)
+  const { realm, realmInfo, config, allowDiscussion } = useRealm()
+  const { refetchProposals } = useWalletStore((s) => s.actions)
+  const isNftPlugin =
+    config?.account.communityTokenConfig.voterWeightAddin &&
+    nftPluginsPks.includes(
+      config?.account.communityTokenConfig.voterWeightAddin?.toBase58()
+    )
+  const { closeNftVotingCountingModal } = useNftProposalStore.getState()
```
Comment (asktree) [inline components/VotePanel/CastVoteButtons.tsx:79]: im confused about whats going on here, we are closing this modal but when does it get opened?

File: components/VotePanel/CastVoteButtons.tsx (hunk 3 above)
Comment (asktree) [reply components/VotePanel/CastVoteButtons.tsx:104]: or alternatively prop-drill it

File: components/VoteCommentModal.tsx [hunk 8]
```diff
@@ -142,6 +140,14 @@ const VoteCommentModal: FunctionComponent<VoteCommentModalProps> = ({
 
   const voteString = VOTE_STRINGS[vote]
 
+  useEffect(() => {
```
Comment (asktree) [inline components/VoteCommentModal.tsx:143]: I think this is probably too hacky; I really don't like it if side effects are used for business logic etcetera like this. 
What I'd rather see is the `submitVote` callback extracted to a hook and used by both this and the parent component, or simply passed as a prop to the modal by the parent.

**Review by asktree — CHANGES_REQUESTED** [review]
see comment

File: components/VoteCommentModal.tsx (hunk 8 above)
Comment (bryzettler) [reply components/VoteCommentModal.tsx:143]: fair, ill work on extracting the logic somehow

**Review by asktree — APPROVED** [review]
great, thanks

**Pull request merged**